d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -281,6 +297,24 @@ private void handleResourceUpdate(Map<String, ? extends ResourceUpdate> updates,
     }
   }
 
+  void handleResourceRemoval(List<String> removedResources, ResourceType type)
+  {
+    if (removedResources == null || removedResources.isEmpty())
+    {
+      return;
+    }
+    for (String resourceName : removedResources)
+    {
+      _xdsClientJmx.incrementResourceNotFoundCount();
+      _log.warn("Received response that resource {} is removed", resourceName);
```

Comment:
Can you log the resource type here as well please?

Reply:
Done